### PR TITLE
Updated readme with all possible vscode preferences

### LINKED ISSUE #40: Document all available preferences in README
... also mention the incompatibility between autoclose tag and brackets

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,22 @@
...
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Allow user to view RPC communication through Output tab.
+* `xml.catalogs` : Indicate XML catalog files
+* `xml.logs.client` : Enable/disable logging to client output.
+* `xml.fileAssociations` : Allow user to associate an XML file pattern with a XML Schema to use.
+* `xml.format.splitAttributes` : A tag's attributes get formatted onto multiple lines.
+* `xml.format.joinCDATALines` : A CDATA tag's content will be normalized.
```
Comment (fbricon) [inline README.md]: Set to `true` to join lines in CDATA content. Defaults to `false`.

File: README.md [hunk 2]
```diff
@@ -40,6 +40,22 @@
...
+* `xml.format.joinContentLines` : An Element's text content will be normalized.
```
Comment (fbricon) [inline README.md]: Set to `true` to join lines in node content during formatting. Defaults to `false`.

File: README.md [hunk 3]
```diff
@@ -40,6 +40,22 @@
...
+* `xml.format.joinCommentLines` : A Comment tag's content will be normalized.
```
Comment (fbricon) [inline README.md]: Set to `true` to join lines in comments during formatting. Defaults to `false`.

File: README.md [hunk 4]
```diff
@@ -40,6 +40,22 @@
...
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Allow user to view RPC communication through Output tab.
+* `xml.catalogs` : Indicate XML catalog files
+* `xml.logs.client` : Enable/disable logging to client output.
+* `xml.fileAssociations` : Allow user to associate an XML file pattern with a XML Schema to use.
+* `xml.format.splitAttributes` : A tag's attributes get formatted onto multiple lines.
```
Comment (fbricon) [inline README.md]: Set to `true` to split node attributes onto multiple lines during formatting. Defaults to `false`.

File: README.md [hunk 5]
```diff
@@ -40,6 +40,22 @@
...
   The Language Server for this client can be [found here](https://github.com/angelozerr/lsp4xml).
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Allow user to view RPC communication through Output tab.
+* `xml.catalogs` : Indicate XML catalog files
+* `xml.logs.client` : Enable/disable logging to client output.
+* `xml.fileAssociations` : Allow user to associate an XML file pattern with a XML Schema to use.
```
Comment (fbricon) [inline README.md]: associate XML Schemas to XML file patterns.

File: README.md [hunk 6]
```diff
@@ -40,6 +40,22 @@
 
   The Language Server for this client can be [found here](https://github.com/angelozerr/lsp4xml).
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Allow user to view RPC communication through Output tab.
+* `xml.catalogs` : Indicate XML catalog files
+* `xml.logs.client` : Enable/disable logging to client output.
```
Comment (fbricon) [inline README.md]: Enable/disable logging to output view.

File: README.md [hunk 7]
```diff
@@ -40,6 +40,22 @@
 
   The Language Server for this client can be [found here](https://github.com/angelozerr/lsp4xml).
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Allow user to view RPC communication through Output tab.
+* `xml.catalogs` : Indicate XML catalog files
```
Comment (fbricon) [inline README.md]: Register XML catalog files.

File: README.md [hunk 8]
```diff
@@ -40,6 +40,22 @@
 
   The Language Server for this client can be [found here](https://github.com/angelozerr/lsp4xml).
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Allow user to view RPC communication through Output tab.
```
Comment (fbricon) [inline README.md]: Traces the communication between VS Code and the XML language server in the Output view.

**Comment by NikolasKomonen** [comment]
@fbricon updated

File: README.md [hunk 9]
```diff
@@ -40,6 +40,22 @@
 
   The Language Server for this client can be [found here](https://github.com/angelozerr/lsp4xml).
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Traces the communication between VS Code and the XML language server in the Output view.
```
Comment (fbricon) [inline README.md]: Trace

File: README.md [hunk 10]
```diff
@@ -40,6 +40,22 @@
...
+* `xml.catalogs` : Register XML catalog files.
+* `xml.logs.client` : Enable/disable logging to output view.
+* `xml.fileAssociations` : Allow user to associate XML Schemas to XML file patterns.
```
Comment (fbricon) [inline README.md]: Associate XML Schemas to XML file patterns.

File: README.md [hunk 11]
```diff
@@ -40,6 +40,22 @@
 
   The Language Server for this client can be [found here](https://github.com/angelozerr/lsp4xml).
 
+## Supported VS Code settings
+
+The following settings are supported:
+
+* `xml.trace.server` : Traces the communication between VS Code and the XML language server in the Output view.
+* `xml.catalogs` : Register XML catalog files.
+* `xml.logs.client` : Enable/disable logging to output view.
```
Comment (fbricon) [inline README.md]: Output

**Pull request merged**